src/main/java/javaslang/collection/HashMap.java

```diff
@@ -440,9 +440,15 @@ public int hashCode() {
     public boolean equals(Object o) {
         if (o == this) {
             return true;
-        } else if (o instanceof HashSet) {
-            final HashSet<?> that = (HashSet<?>) o;
-            return this.iterator().equals(that.iterator());
+        } else if (o instanceof HashMap) {
+            final Iterator<?> thisIt = this.iterator();
```

Comment:
Let's change `Iterator.equals(Iterator<? extends T> that)` to `Iterator.equals(Iterator<?> that)` and then

```
final HashMap<?, ?> that = (HashMap<?, ?>) o;
return this.iterator().equals(that.iterator())`?
```

this reduces duplicate code and may be also used in other collections.